### DISCUSSION TITLE
harmonic number autodetect

### DISCUSSION THREAD

**Discussion by fabiomancini67** [GrandOrgue/GoOdf#91] [Ideas]
It would be very useful to have the autodetect function for the harmonic number as it was present in the Organ Builder software. This function is particularly important in the case of Mixtures, or other Stops with unknown and unconventional textures such as cornets and similar.

**Comment by larspalo** [comment]
It already exists. If you're absolutely sure that the pitch info in the first loaded pipe sample is correct and you know the reference pitch of a1, you can try hitting the button to the left of the harmonic number that's named "Calculate HN" and see if that does the trick for you.

It's rather pointless for stops/ranks that never change (break back) as it's faster to just change the number for the whole rank, but for mixtures and other compound stops that need to have the HN set on the pipe level it's quite nice.

**Reply by larspalo** [reply]
Word of caution. One should be sure that the pitch information in the first sample of _every_ pipe for the whole rank is correct before using the "Calculate HN" button.

**Comment by fabiomancini67** [comment]
Hello Lars, thank you very much for pointing me at that function!
I just tested it and it works well with cornets and mixtures, but it doesn't work with cymbales, I think because the pitch of a1 is higher then 880 Hz, the maximum allowed, in fact it is around 1574 Hz.

**Reply by larspalo** [reply]
The pitch is expected to be given as reference pitch of what is used in the instrument and what can be expected for a sample of 8' pitch tone a1 (069-A). For example in the Norrfjärden Sample Set I would need to set it to 468 or something like that with this feature, no matter what type of compound stop it is used for. But it's absolutely essential that the pitch info embedded in the files is correct or else issues will arise. Also I think the detection algorithm could be improved as it's currently perhaps too sensitive.

**Reply by fabiomancini67** [reply]
Ok thank you, now I understood, it's much simpler then expected. I just checked and it works! There are just a few notes mistuned whit the cymbales when a teperament is applied in Grandorgue, I have to understand why, I will investigate in the samples. I'm currently trying to play the Prytanee organ in Grandorgue, it is tuned with a1 at 393.4 Hz
